framework/sandstone_opts.cpp

```diff
@@ -933,7 +824,427 @@ int parse_cmdline(int argc, char** argv, SandstoneApplication* app, ParsedCmdLin
                 return EX_USAGE;
             }
         }
-    } else {
+        return EXIT_SUCCESS;
+    }
+
+    // validate dynamic conditions, like other conflicting arguments' presence
+    int validate_args() const
```

Comment:
We had a discussion with @thiagomacieira about this. We cannot introduce breaking changes (e.g. customer scripts based on existing behavior). At most, we can warn about certain conditions. As such, argument validation cannot return `EX_USAGE` for any checks that were newly introduced. I think warning is OK, but not termination. Please make sure that the warning message is not too harsh, i.e. instead of "a and b are mutually exclusive" use "a and b were specified together, using <value>". Alternatively, we can just eliminate `validate_args()` altogether from this changeset and introduce them later.